flickit-assessment-kit/src/main/java/org/flickit/assessment/kit/adapter/out/persistence/kitversion/KitVersionPersistenceJpaAdapter.java

```diff
@@ -38,14 +38,15 @@ public long persist(CreateKitVersionPort.Param param) {
         var kitEntity = kitRepository.findById(param.kitId())
             .orElseThrow(() -> new ResourceNotFoundException(KIT_ID_NOT_FOUND));
 
-        var versionEntity = KitVersionMapper.createParamToJpaEntity(kitEntity, param);
-        versionEntity.setId(sequenceGenerators.generateKitVersionId());
+        var id = sequenceGenerators.generateKitVersionId();
+        var versionEntity = KitVersionMapper.createParamToJpaEntity(id, kitEntity, param);
         return repository.save(versionEntity).getId();
     }
 
     @Override
     public void updateStatus(long kitVersionId, KitVersionStatus newStatus) {
-        repository.updateStatus(kitVersionId, newStatus.getId());
+        long statusVersion = newStatus.getId() == 2 ? -kitVersionId : newStatus.getId();
+        repository.updateStatus(kitVersionId, newStatus.getId(), statusVersion);
```

Comment:
**Suggestion:** Add error handling for the new statusVersion logic in updateStatus to manage cases where the statusVersion calculation might lead to unexpected values or database constraints violations. [possible issue, importance: 7]
```suggestion
        try {
            repository.updateStatus(kitVersionId, newStatus.getId(), statusVersion);
        } catch (Exception e) {
            // Handle exception, possibly logging or re-throwing
        }
```